packages/transaction-controller/src/helpers/GasFeePoller.ts

```diff
@@ -0,0 +1,178 @@
+   */
+  constructor({
+    gasFeeFlows,
+    getChainIds,
+    getEthQuery,
+    getGasFeeControllerEstimates,
+    getTransactions,
+    onStateChange,
+  }: {
+    gasFeeFlows: GasFeeFlow[];
+    getChainIds: () => Hex[];
+    getEthQuery: () => EthQuery;
+    getGasFeeControllerEstimates: () => Promise<GasFeeState>;
+    getTransactions: () => TransactionMeta[];
+    onStateChange: (listener: () => void) => void;
+  }) {
+    this.#gasFeeFlows = gasFeeFlows;
+    this.#getChainIds = getChainIds;
+    this.#getEthQuery = getEthQuery;
+    this.#getGasFeeControllerEstimates = getGasFeeControllerEstimates;
+    this.#getTransactions = getTransactions;
+
+    onStateChange(() => {
+      const unapprovedTransactions = this.#getUnapprovedTransactions();
+
+      if (unapprovedTransactions.length) {
+        this.#start();
+      } else {
+        this.#stop();
+      }
+    });
+  }
+
+  #start() {
+    if (this.#running) {
+      return;
+    }
+
+    // Intentionally not awaiting since this starts the timeout chain.
+    // eslint-disable-next-line @typescript-eslint/no-floating-promises
+    this.#onTimeout();
+
+    this.#running = true;
+
+    log('Started polling');
+  }
+
+  #stop() {
+    if (!this.#running) {
+      return;
+    }
+
+    clearTimeout(this.#timeout);
+
+    this.#timeout = undefined;
+    this.#running = false;
+
+    log('Stopped polling');
+  }
+
+  async #onTimeout() {
+    await this.#updateUnapprovedTransactions();
+
+    // eslint-disable-next-line @typescript-eslint/no-misused-promises
+    this.#timeout = setTimeout(() => this.#onTimeout(), INTERVAL_MILLISECONDS);
+  }
+
+  async #updateUnapprovedTransactions() {
+    const unapprovedTransactions = this.#getUnapprovedTransactions();
+
+    log('Found unapproved transactions', {
+      count: unapprovedTransactions.length,
+    });
+
+    const ethQuery = this.#getEthQuery();
+
+    await Promise.all(
+      unapprovedTransactions.map((tx) =>
+        this.#updateTransactionSuggestedFees(tx, ethQuery),
+      ),
+    );
+  }
+
+  async #updateTransactionSuggestedFees(
+    transactionMeta: TransactionMeta,
+    ethQuery: EthQuery,
+  ) {
+    const gasFeeFlow = getGasFeeFlow(transactionMeta, this.#gasFeeFlows);
+
+    if (!gasFeeFlow) {
+      log('Skipping update as no gas fee flow found', transactionMeta.id);
+
+      return;
+    }
+
+    log('Found gas fee flow', gasFeeFlow.constructor.name, transactionMeta.id);
+
+    const request: GasFeeFlowRequest = {
+      ethQuery,
+      getGasFeeControllerEstimates: this.#getGasFeeControllerEstimates,
+      transactionMeta,
+    };
+
+    try {
+      const response = await gasFeeFlow.getGasFees(request);
+
+      transactionMeta.gasFeeEstimates = response.estimates;
+    } catch (error) {
+      log('Failed to get suggested gas fees', transactionMeta.id, error);
+      return;
+    }
+
+    this.hub.emit(
+      'transaction-updated',
+      transactionMeta,
+      'GasFeePoller - Suggested gas fees updated',
+    );
```

Comment:
This will update the history object when the suggested fees are changed. 
So if user leave transaction `unapproved` in any mainnet (where suggestions change very often), it may get bigger and lead some issues.

Reply:
Good point. I've skipped the history update specifically from `GasFeePoller` to avoid bloating the state.